src/plugins/intel_cpu/src/cpu_streams_calculation.cpp

```diff
@@ -52,23 +58,104 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
         }
     };
 
+    auto update_ids_method = [&](const std::vector<int>& one_proc_info) {
+        stream_info[STREAM_NUMA_NODE_ID] = one_proc_info[PROC_NUMA_NODE_ID];
+        stream_info[STREAM_SOCKET_ID] = one_proc_info[PROC_SOCKET_ID];
+    };
+
+    auto update_streams_per_node = [&](const int& proc_type, const std::vector<int>& one_proc_info) {
+        if (0 != one_proc_info[proc_type]) {
+            if (n_threads_per_stream == -1) {
+                stream_info[THREADS_PER_STREAM] = (proc_type == EFFICIENT_CORE_PROC) ? 2 : 1;
+            }
+            stream_info[PROC_TYPE] = proc_type;
+            update_ids_method(one_proc_info);
+            stream_info[NUMBER_OF_STREAMS] =
+                static_cast<int>(one_proc_info[proc_type] / stream_info[THREADS_PER_STREAM]);
+            if ((stream_info[NUMBER_OF_STREAMS] == 0) && (proc_type == MAIN_CORE_PROC)) {
+                stream_info[NUMBER_OF_STREAMS] =
+                    static_cast<int>((one_proc_info[MAIN_CORE_PROC] + one_proc_info[HYPER_THREADING_PROC]) /
+                                     stream_info[THREADS_PER_STREAM]);
+            }
+            if (n_streams <= stream_info[NUMBER_OF_STREAMS]) {
+                stream_info[NUMBER_OF_STREAMS] = n_streams;
+                streams_info_table.push_back(stream_info);
+            } else {
+                streams_info_table.push_back(stream_info);
+            }
+            n_streams -= stream_info[NUMBER_OF_STREAMS];
+            proc_socket_table[one_proc_info[PROC_SOCKET_ID]][proc_type] -=
```

Comment:
It failed in this case.
set "numactl -C" in two socket machine which enabled socket_id = 1 and no socket_id = 0.
proc_type_table.size = 1, proc_socket_table,size = 1 in this case.
so it can cause access out of bounds.

Reply:
It is resolved in PR18264.